duecredit/io.py

```diff
@@ -12,36 +12,45 @@
 if 'DUECREDIT_TEST_EARLY_IMPORT_ERROR' in os.environ.keys():
     raise ImportError("DUECREDIT_TEST_EARLY_IMPORT_ERROR")
 
-import re
-import locale
-import time
-from collections import defaultdict
 import copy
-from os.path import dirname, exists
+import locale
 import pickle
+import re
+import sys
 import tempfile
+import time
 import warnings
-import platform
-from time import sleep
+from collections import defaultdict
+from distutils.version import StrictVersion
+from os.path import dirname, exists
+from typing import Any, Dict, List, Optional, Tuple, Type, Union, TYPE_CHECKING
 
 from .config import CACHE_DIR, DUECREDIT_FILE
 from .entries import BibTeX, Doi, Text, Url
 from .log import lgr
 from .versions import external_versions
 
+if TYPE_CHECKING:
+    from collector import Citation
+
 _PREFERRED_ENCODING = locale.getpreferredencoding()
-platform_system = platform.system().lower()
-on_windows = platform_system == 'windows'
+
+# https://github.com/python/mypy/issues/9242#issuecomment-667586397
+# https://github.com/spack/spack/pull/35640
+if sys.platform == "win32":
+    OSExceptions = (OSError, WindowsError)
```

Comment:
`WindowsError` has been a subclass of `OSError` since Python 3.3.  Just use `OSError` by itself.